src/librustc_trans/abi.rs

```diff
@@ -133,32 +134,279 @@ impl ArgAttributes {
     }
 }
 
+pub trait CastTarget {
```

Comment:
Any reason this is a trait rather than an enum? That would make moving away from LLVM easier.